jupyterhub_traefik_proxy/toml.py

```diff
@@ -162,10 +180,10 @@ def get_target_data(d, to_find):
         The proxy implementation should also have a way to associate the fact that a
         route came from JupyterHub.
         """
+        routespec = self.validate_routespec(routespec)
 
-        self.log.info("Adding route for %s to %s.", routespec, target)
+        self._sync_cache_if_empty()
```

Comment:
Maybe `traefik_utils.load_routes()` should be called exactly once in `__init__`?

```python
def __init__(self, **kwargs):
    super().__init__()
    # load initial routing table from disk
    try:
        self.routes_cache = traefik_utils.load_routes(...)
    except FileNotFoundError:
        pass

Reply:
You're right, I've over-complicated it.
However, I believe we still need to initialize the routes_cache with the *frontends* and *backends* keys.
We also need to check if the file is empty and if it is, initialize with the keys above.
Something like this:

```python
def __init__(self, **kwargs):
        super().__init__(**kwargs)
        try:
            # Load initial routing table from disk
            self.routes_cache = traefik_utils.load_routes(
                self.toml_dynamic_config_file
            )
        except FileNotFoundError:
            self.routes_cache = {}
        finally:
            if not self.routes_cache:
                self.routes_cache = {"backends": {}, "frontends": {}}
```
What do you think?